enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/document/DocumentVerificationService.java

```diff
@@ -105,14 +116,80 @@ public void executeFinalDocumentVerification(final IdentityVerificationEntity id
+        moveIdentityVerificationToDocumentVerificationFinal(result, identityVerification, ownerId);
+    }
+
+    /**
+     * Starts the verification process
+     *
+     * @param ownerId Owner identification.
+     * @param identityVerification Identity verification.
+     * @throws RemoteCommunicationException In case of remote communication error.
+     * @throws DocumentVerificationException In case of business logic error.
+     */
+    @Transactional
+    public void startVerification(OwnerId ownerId, IdentityVerificationEntity identityVerification) throws DocumentVerificationException, RemoteCommunicationException, OnboardingProcessException {
+        final List<DocumentVerificationEntity> docVerifications =
+                documentVerificationRepository.findAllDocumentVerifications(identityVerification, List.of(DocumentStatus.VERIFICATION_PENDING));
+
+        final List<DocumentVerificationEntity> selfiePhotoVerifications = docVerifications.stream()
+                        .filter(entity -> entity.getType() == DocumentType.SELFIE_PHOTO)
+                        .collect(toList());
+
+        // If not enabled then remove selfie photos from the verification process
+        if (!identityVerificationConfig.isVerifySelfieWithDocumentsEnabled()) {
+            docVerifications.removeAll(selfiePhotoVerifications);
+        }
+
+        documentProcessingService.pairTwoSidedDocuments(docVerifications);
+
+        List<String> uploadIds = docVerifications.stream()
+                .map(DocumentVerificationEntity::getUploadId)
+                .collect(toList());
+
+        final DocumentsVerificationResult result = documentVerificationProvider.verifyDocuments(ownerId, uploadIds);
+        final String verificationId = result.getVerificationId();
+        final DocumentVerificationStatus status = result.getStatus();
+        logger.info("Verified documents upload ID: {}, verification ID: {}, status: {}, {}", uploadIds, verificationId, status, ownerId);
+        auditService.auditDocumentVerificationProvider(identityVerification, "Documents verified: {} for user: {}", status, ownerId.getUserId());
+
+        if (!identityVerificationConfig.isVerifySelfieWithDocumentsEnabled()) {
+            logger.debug("Selfie photos verification disabled, changing selfie document status to ACCEPTED, {}", ownerId);
+            selfiePhotoVerifications.forEach(selfiePhotoVerification -> {
+                selfiePhotoVerification.setStatus(DocumentStatus.ACCEPTED);
+                selfiePhotoVerification.setTimestampLastUpdated(ownerId.getTimestamp());
+                auditService.audit(selfiePhotoVerification, "Selfie document accepted for user: {}", ownerId.getUserId());
+            });
+            documentVerificationRepository.saveAll(selfiePhotoVerifications);
+        }
+
+        changeDocumentVerificationStatusAndVerificationId(docVerifications, result, identityVerification, ownerId);
```

Comment:
The main reason of change is here. We change the status of the documents based on the response from the provider. It used to be (whatever the result was)
```java
docVerifications.forEach(docVerification -> {
            docVerification.setStatus(DocumentStatus.VERIFICATION_IN_PROGRESS);
```

Reply:
🤔 May be this is too complicated and calling `verificationProcessingService.processVerificationResult` could be enough.